sqlglot/generator.py

```diff
@@ -77,6 +77,7 @@ class Generator:
         exp.ExecuteAsProperty: lambda self, e: self.naked_property(e),
         exp.ExternalProperty: lambda self, e: "EXTERNAL",
         exp.HeapProperty: lambda self, e: "HEAP",
+        exp.Heredoc: lambda self, e: f"${e.args['tag'] or ''}$\n{e.this}\n${e.args['tag'] or ''}$",
```

Comment:
make this e.args("tag") and move this to a first class function

def heredoc_sql ...